### DISCUSSION TITLE
Has anyone succeeded in using Sqitch in a Github Action?

### DISCUSSION THREAD

**Discussion by hanszoons** [theory/sqitch#794] [Q&A]
I am unable to use Sqitch within a Github Action. I have tried using Docker through the 'container' option of actions, directly with the commands as described on the sqitch docker, through both methods of installation on Ubuntu/Debian (https://sqitch.org/download/debian/), but it just doesn't work to get sqitch to do its job as it does locally.

Additionally, I want to work with environment variables to address a different database server within the github action than on the production server, but of course I do not want to hard-code the connection string in sqitch.conf. That too turns out to be no easy task.

Even when I manage to get Sqitch running, the scripts also don't seem really suited for a github action, as even a simple sqitch status results in a Error: Process completed with exit code 1. and ends the action (while it should just pass if there is no work for Sqitch).

This really determines whether we can use Sqitch in our workflow or need to look for another solution. Does anyone perhaps have successful action .yml's available that I can learn from?

**Comment by theory** [comment]
What issues did you run into? That error doesn't tell us much. Is it failing to connect to the database? Does it work properly outside of GitHub Actions?

You should be able to set environment variables conditionally, no? Can you share your workflow YAML file?
